src/test/java/seedu/address/logic/commands/EditTaskCommandTest.java

```diff
@@ -1,13 +1,111 @@
 package seedu.address.logic.commands;
 
-//import static org.junit.jupiter.api.Assertions.assertFalse;
-//import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.logic.commands.CommandTestUtil.showPersonAtIndex;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PERSON;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_TASK;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PERSON;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
 
 import org.junit.jupiter.api.Test;
 
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Patient;
+import seedu.address.model.task.Task;
+import seedu.address.testutil.PersonBuilder;
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for EditTaskCommand.
+ */
 class EditTaskCommandTest {
 
+    private static final String TASK_STUB = "Some task";
+
+    private final Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_editTask_success() {
+        Task editedTask = new Task(TASK_STUB);
+
+        // Use second patient as the first patient in typical persons does not have a task
+        Patient secondPatient = model.getFilteredPersonList().get(INDEX_SECOND_PERSON.getZeroBased());
+        Patient editedPatient = new PersonBuilder(secondPatient)
+                .withTasks(secondPatient.getTasks().edit(INDEX_FIRST_TASK.getZeroBased(), editedTask)
+                        .getTasks().stream().map(Object::toString).toArray(String[]::new)) // to convert the TaskList
+                // into an array of string representation of tasks
```

Comment:
Would it be better to extract the conversion of the `TaskList` to the array of strings as a method in the `TaskList` class?

Reply:
The necessity for an array of strings is very specific to these testing cases right (for the withTasks), so I am not sure where it belongs.